File: C++/longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char, int> letters;
        for (char c : s)
        {
            letters[c]++;
        }

        int ans = 0;

        int maxOdd = 0;
        char oddChar = '\0';

        for (auto it : letters)
        {
            char c = it.first;
            int n = it.second;

            if (n % 2 == 0)
            {
                ans += n;
            }
            else
            {
                if (n > 2)
                {
                    ans += n - 1;
                }

                if (n > maxOdd)
                {
                    maxOdd = n;
                    oddChar = c;
                }
            }
        }

        if (maxOdd)
        {
            ans += 1;
        }

        return ans;
    }
};

File: Go/longest-palindrome.go
func longestPalindrome(s string) int {
	letters := map[byte]int{}
	for _, c := range []byte(s) {
        letters[c]++
	}
    ones := 0
    size := 0
    for _, count := range letters {
        if count % 2 == 0 {
            size += count
        } else {
            size += count / 2 * 2
            ones++
        }
    }
    if ones > 0 {
        size++
    }
	return size
}